src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/UserService.java

```diff
@@ -101,6 +105,8 @@ public List<User> getUsers() {
     public void deleteUser(User user) {
         log.debug("deleting user with id: {}", user.getId());
 
+        applicationRepository.deleteAll(applicationRepository.findByApplicant_Id(user.getId()));
```

Comment:
you can use `void deleteAllByApplicant_Id(@NotNull UUID id);` when added to the repository

Reply:
oh wow, that's pretty cool